### ISSUE TITLE
Node API does not support plugins

### ISSUE THREAD

**Issue by danez** [eslint/eslint#6841]
**What version of ESLint are you using?**
3.2.2

**What parser (default, Babel-ESLint, etc.) are you using?**
default, babel-eslint, doesn't matter

**Please show your full configuration:**
```js
var config = {
    parser: require.resolve(".."),
    plugins: [
      "flow-vars"
    ],
    rules: Object.assign({}, rules, {
      "flow-vars/define-flow-type": 1,
      "flow-vars/use-flow-type": 1
    }),
    env: {
      node: true,
      es6: true
    },
    parserOptions: {
      ecmaVersion: 6,
      ecmaFeatures: {
        jsx: true,
        experimentalObjectRestSpread: true,
        globalReturn: true
      },
      sourceType: sourceType
    }
  };
```

**What did you do? Please include the actual source code causing the issue.**

```var messages = eslint.linter.verify(code, config);````

**What did you expect to happen?**

linting warnings or no warnings

**What actually happened? Please include the actual, raw output from ESLint.**

`Definition for rule 'flow-vars/define-flow-type' was not found`

---

After digging into `verify` it does not handle or apply plugins (it simply drops it in prepareConfig), so even if I throw an error in the plugin or add a plugin that does not exist, nothing happens.

We are trying to add tests to babel-eslint (babel/babel-eslint#361) that use `verify()` and also have plugins applied.

**Comment by platinumazure** [comment]
You'll want to use CLIEngine.executeOnFiles(). The linter's sole job is to lint files with a prepared configuration.

**Comment by mysticatea** [comment]
Yes. Please use CLIEngine in that case. `linter` is API which should work in browsers, so it cannot depend on file system.

**Comment by danez** [comment]
I see thanks, thats what we ended up doing.
